src/java/org/apache/cassandra/utils/obs/OffHeapBitSet.java

```diff
@@ -51,15 +51,9 @@ public OffHeapBitSet(long numBits, MemoryLimiter memoryLimiter) throws MemoryLim
         long wordCount = (((numBits - 1) >>> 6) + 1);
         if (wordCount > Integer.MAX_VALUE)
             throw new UnsupportedOperationException("Bloom filter size is > 16GB, reduce the bloom_filter_fp_chance");
-        try
-        {
-            long byteCount = wordCount * 8L;
-            bytes = allocate(byteCount, memoryLimiter);
-        }
-        catch (OutOfMemoryError e)
-        {
-            throw new RuntimeException("Out of native memory occured, You can avoid it by increasing the system ram space or by increasing bloom_filter_fp_chance.");
```

Comment:
we are losing this hint for the system administrator, I would keep it, and maybe add `e` as "initCause" of the RuntimeException

Reply:
It's handled with very similar message here: https://github.com/datastax/cassandra/blob/main/src/java/org/apache/cassandra/utils/FilterFactory.java#L90

Reply:
Also this code seemed weird since here we catch OOM and throw RuntimeException, but invoking function is catching OOM and not RuntimeException 😕

Reply:
can yu please change the error message in the OutOfMemory exception, keeping the reference to the "bloom_filter_fp_chance" parameter ? this may help the people who is tourbleshooting and solving problems in production

Reply:
Would it be useful to store the information about the sizes of data we couldn't allocate, that is, use something more sophisticated than a counter?

Reply:
>Would it be useful to store the information about the sizes of data we couldn't allocate, that is, use something more sophisticated than a counter?

I don't think there is appropriate/meaningful way to store that in Prometheus so we probably need look that up in the logs.